braindecode/datasets/tuh.py

```diff
@@ -123,21 +128,37 @@ def _parse_description_from_file_path(file_path):
     #                          subject/recording session/file
     # e.g.                 tuh_eeg/v1.1.0/edf/01_tcp_ar/027/00002729/
     #                          s001_2006_04_12/00002729_s001.edf
-    version = tokens[-7]
-    year, month, day = tokens[-2].split('_')[1:]
-    subject_id = tokens[-3]
+    if 'train' or 'eval' in tokens:
+        version = tokens[-6]
+    else:
+        version = tokens[-7]
+    subject_id = tokens[-1].split('_')[-2].split('s')[-1]
```

Comment:
Is this correct? The new subject identifier is a combination of letters.
e.g. 'aaaaaaav_s004_t000.edf'

From the readme:

```
The last segment is the filename ("aaaaamye_s001_t000.edf"). This
includes the subject identifier ("aaaaamye"), the session number
("s001") and a token number ("t000"). EEGs are split into a series of
files starting with *t000.edf, *t001.edf, ...  These represent pruned
EEGs, so the original EEG is split into these segments, and
uninteresting parts of the original recording were deleted (common in
clinical practice).
```

To make it work on my side, I avoided conversion to integer downstream.